public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx

```diff
@@ -267,8 +268,9 @@ export class VizPanelManager extends SceneObjectBase<VizPanelManagerState> {
     };
 
     const newOptions = newPlugin?.onPanelTypeChanged?.(panel, prevPluginId, prevOptions, prevFieldConfig);
+
     if (newOptions) {
-      newPanel.onOptionsChange(newOptions, true);
+      newPanel.onOptionsChange(newOptions, true, true);
```

Comment:
I doesn't fix the issue, since the `newPlugin` is never there, because the loading of the plugin is async.

This will need a bigger refactor, to either move all this logic to "changeType" function in `VizPanel` or add an state handler to trigger when the plugin is loaded. 

I'd go for the first approach but it will need a big refactor here.

@kaydelaney do you have any better ideas?